src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1397,104 +1408,115 @@ private int onDecodeConnect(
 
                 if (sessionExpiryInterval > 0)
                 {
-                    final int flags = connect.flags();
-                    final String topic = String.format(SESSION_WILDCARD_TOPIC_FORMAT, clientId);
-                    final int topicKey = topicKey(topic);
-                    final RouteFW route = resolveTarget(routeId, authorization, topic,  PUBLISH_ONLY);
-
-                    if (route != null)
+                    progress = onResolveSession(traceId, authorization, reasonCode, progress, clientIdentifier, connect, payload);
+                }
+                else
+                {
+                    if (reasonCode == SUCCESS)
                     {
-                        final long resolvedId = route.correlationId();
+                        doCancelConnectTimeoutIfNecessary();
+                        doEncodeConnack(traceId, authorization, reasonCode, clientIdentifier.value());
+
+                        connected = true;
```

Comment:
Are we using `connected` to mean two different things?
 - already received `CONNECT`
 - already sent `CONNACK`

Previously, when `onDecodeConnect` called `doEncodeConnack` directly, these were effectively the same, but now that `doEncodeConnack` is asynchronous, we may no longer be properly detecting a second `CONNECT` frame arriving before we have sent `CONNACK`.